app/Http/Controllers/ClientController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ClientStoreRequest;
+use App\Models\Client;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Auth;
+
+class ClientController extends Controller
+{
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     *
+     */
+    public function index()
+    {
+        $clients = Client::withCount('projects')->with('projects')->orderBy('projects_count', 'desc')->paginate(10);
+        return view('clients.index', compact('clients'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        abort_if(!Auth::user()->hasRole('admin'), 401, __('Not Allowed action'));
```

Comment:
@faxunil technically, 401 code means "unauthenticated", which means not logged in. In this case, it should be 403, meaning "unauthorized" or "logged in but has no permission".